pkg/flowdebug/flowdebug.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flowdebug
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+var perFlowDebug = false
+
+// Enable enables per-flow debugging
+func Enable() {
+	perFlowDebug = true
+}
+
+// Log must be used to log any debug messages emitted per request/message
+func Log(l *logrus.Entry, msg string) {
+	if perFlowDebug {
```

Comment:
Instead of testing a flag for every debug statement, why not creating a new logger with a log level that depends on the command line option, only once when the process starts?

Reply:
What's the difference? Do you want to micro-optimize a debug call? :-)

Reply:
No, I thought that would be cleaner.